Engine/DataFeeds/Enumerators/SubscriptionDataEnumerator.cs

```diff
@@ -67,6 +73,13 @@ public bool MoveNext()
             var result = _enumerator.MoveNext();
             if (result)
             {
+                // Check if this subscription current data is auxiliary and if it is, verify that it should emit it
+                if (_enumerator.Current != null && _enumerator.Current.DataType == MarketDataType.Auxiliary && !ShouldEmitAuxiliaryData)
+                {
+                    // We shouldn't emit this data, so we will MoveNext() again.
+                    return MoveNext();
```

Comment:
hm sure about moving next here? We are not using `_enumerator.Current` 🤔 
Why is `ShouldEmitAuxiliaryData` public?

Reply:
Yeah I used `MoveNext()` there because we are explicitly skipping `_enumerator.Current` because it is Aux data that shouldn't be emitted. That way we just move right past it without the caller knowing any better. 

Its like in `SubscriptionUtils.CreateAndScheduleWorker` where I continue if the data is aux and we shouldn't emit, it just repeats the loop and MovesNext() again effectively skipping the aux data.

Reply:
Ill make `ShouldEmitAuxiliaryData` private readonly 👍🏽